frontend-react/src/components/USLink.tsx

```diff
@@ -17,25 +16,51 @@ type USLinkProps = AnchorHTMLAttributes<{}> &
     Omit<CustomLinkProps, "activeClassName">;
 type USNavLinkProps = Pick<AnchorHTMLAttributes<{}>, "href"> & CustomLinkProps;
 
+/**
+ * Stateless function to get route href from href that could be
+ * absolute, shorthand, relative, and/or a non-route.
+ * Attempt to parse href as URL (taking into account "//" shorthand).
+ * If it errors, then assume its a relative url (aka route). If it
+ * parses, then verify its an absolute route through origins.
+ */
+export function getHrefRoute(href?: string): string | undefined {
+    if (href === undefined) return undefined;
+
+    try {
+        const url = new URL(
+            href.replace(/^\/\//, `${window.location.protocol}//`)
+        );
+        if (
+            url.protocol.startsWith("http") &&
+            url.origin === window.location.origin
+        )
+            return `${url.pathname}${url.search}`;
+    } catch (e: any) {
+        return href;
+    }
+
+    return undefined;
+}
+
 /** A single link for rendering standard links. Uses a `Link` by default
  * but adding `anchor` will make this a generic anchor tag.
  * @example
  * <USLink href="/page">To Page</USLink> // uses <Link> from react-router-dom
- * <USLink anchor href="#this-section-on-my-page">To Section</USLink> // uses <a>
+ * <USLink href="#this-section-on-my-page">To Section</USLink> // uses <a>
  * */
 export const USLink = ({
-    anchor = false,
     children,
     className,
     href,
     state,
     ...anchorHTMLAttributes
 }: USLinkProps) => {
     const sanitizedHref = href ? DOMPurify.sanitize(href) : href;
+    const routeHref = getHrefRoute(sanitizedHref);
 
-    return !anchor && sanitizedHref ? (
+    return routeHref !== undefined ? (
```

Comment:
VERY PEDANTIC AND SUBJECTIVE POINT, but would you mind changing this to be `return Boolean(routeHref)`?  I feel like it's often easier to reason about with positive booleans, but that's just my simple brain getting confused with booleans 😆

Reply:
The differentiation on undefined is actually important here. An empty string is considered the current page (aka route) but undefined means you're not using a URL (aka not a route) so you should get an anchor. Blind Boolean/negating will consider an empty string as false.

Reply:
Ah, yes -- you're definitely right here!